module/VuFind/src/VuFind/Db/Service/ShortlinksService.php

```diff
@@ -0,0 +1,178 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Service;
+
+use Exception;
+use Laminas\Log\LoggerAwareInterface;
+use VuFind\Db\Entity\Shortlinks;
+use VuFind\Log\LoggerAwareTrait;
+
+/**
+ * Database service for shortlinks.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+class ShortlinksService extends AbstractService implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    /**
+     * Create a shortlinks entity object.
+     *
+     * @return Shortlinks
+     */
+    public function createEntity(): Shortlinks
+    {
+        $class = $this->getEntityClass(Shortlinks::class);
+        return new $class();
+    }
+
+    /**
+     * Generate a short hash using the base62 algorithm (and write a row to the
+     * database).
+     *
+     * @param string $path Path to store in database
+     *
+     * @return string
+     */
+    public function getBase62Hash(string $path): string
+    {
+        $shortlink = $this->createEntity();
+        $now = new \DateTime();
+        $shortlink->setPath($path)
+            ->setCreated($now);
+        try {
+            $this->entityManager->persist($shortlink);
+            $this->entityManager->flush();
+            $id = $shortlink->getId();
+            $b62 = new \VuFind\Crypt\Base62();
+            $shortlink->setHash($b62->encode($id));
+            $this->entityManager->persist($shortlink);
+            $this->entityManager->flush();
+        } catch (\Exception $e) {
+            $this->logError('Could not save shortlink: ' . $e->getMessage());
+            throw $e;
+        }
+        return $shortlink->getHash();
+    }
+
+    /**
+     * Support method for getGenericHash(): do the work of picking a short version
+     * of the hash and writing to the database as needed.
+     *
+     * @param string $path          Path to store in database
+     * @param string $hash          Hash of $path (generated in getGenericHash)
```

Comment:
We could also remove `(generated in getGenericHash)` from here to prevent confusion.